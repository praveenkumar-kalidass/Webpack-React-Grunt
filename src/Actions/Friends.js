import Api from "../Api/Friends";
import actions from "./ActionTypes";
import store from "../Store/store";

const friendsActions = {
  getAllFriends: () => {
    store.dispatch((action) => {
      Api.getAllFriends().then((friends) => {
        action({
          type: actions.GET_ALL_FRIENDS,
          data: friends.data
        });
      }).catch((error) => {
        throw(error);
      });
    });
  }
};

export default friendsActions;
