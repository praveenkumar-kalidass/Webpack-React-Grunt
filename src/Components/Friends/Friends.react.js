import React from "react";
import {connect} from "react-redux";
import {Button} from "react-bootstrap";
import FriendsActions from "../../Actions/Friends";

const mapStateToProps = (state) => (
  {
    friends: state.friends
  }
);

class Friends extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      friends: []
    };
  }

  componentWillReceiveProps(props) {
    this.setState({
      friends: props.friends.friends
    });
  }

  componentDidMount() {
    FriendsActions.getAllFriends();
  }

  render() {
    return (
      <div>
        Friends Page
      </div>
    );
  }
}

export default connect(mapStateToProps)(Friends);
