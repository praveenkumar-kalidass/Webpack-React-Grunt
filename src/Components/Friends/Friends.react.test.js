import React from 'react';
import Friends from './Friends.react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk";
import {Provider} from "react-redux";
import {shallow} from "enzyme";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store;

test("Friends Component", () => {
  store = mockStore();
  const component = renderer.create(
    <Friends store={store} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
