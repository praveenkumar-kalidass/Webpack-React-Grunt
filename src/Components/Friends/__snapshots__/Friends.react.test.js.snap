// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friends Component 1`] = `
<div>
  Friends Page
</div>
`;
